androidApp/src/main/java/com/example/tasky/android/agenda/components/DetailsDeleteSection.kt

```diff
@@ -19,16 +19,23 @@ import com.example.tasky.model.agenda.Task
 @Composable
 fun DetailsDeleteSection(
     item: AgendaItem,
+    eventIsGoing: Boolean,
     onClick: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
     TextButton(onClick = onClick, modifier = modifier) {
         Text(
             stringResource(
-                when (item) {
-                    is Event -> R.string.delete_event
-                    is Task -> R.string.delete_task
-                    is Reminder -> R.string.delete_reminder
+                if (item !is Event || item.isUserEventCreator) {
+                    when (item) {
+                        is Event -> R.string.delete_event
+                        is Task -> R.string.delete_task
+                        is Reminder -> R.string.delete_reminder
+                    }
+                } else if (eventIsGoing) {
+                    R.string.leave_event
+                } else {
+                    R.string.join_event
```

Comment:
I find this logic a bit hard to follow. Personally, I like to use Kotlin’s `when` statement for multiple conditions like this:

```
text = when (agendaDetails) {
  is AgendaItemDetails.Event -> when {
    state.isUserEventCreator -> stringResource(R.string.delete_event).uppercase()
    state.isUserGoing -> stringResource(R.string.leave_event).uppercase()
    else -> stringResource(R.string.join_event).uppercase()
  }

  AgendaItemDetails.Reminder -> stringResource(R.string.delete_reminder).uppercase()
  is AgendaItemDetails.Task -> stringResource(R.string.delete_task).uppercase()
}
```

Again, this is just my preference. Feel free to use `if` statements, but I’d advise you first to see if you can’t create the statement without using negating conditions, as they are harder to follow and reason about.